### ISSUE TITLE
bug: issues with workspaces

### ISSUE THREAD

**Issue by EinfachHans** [raineorshine/npm-check-updates#1312]
- [X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [X] I am using the latest version of `npm-check-updates`
- [X] I am using `node >= 14.14`

---

## Steps to Reproduce

.ncurc:

```js
module.exports = {
  interactive: true,
  format: ['group', 'repo']
}
```

Steps:

`npm-check-updates -ws --root`

## Current Behavior

I have a monorepo with npm workspaces and mentioned the following issues:

1. The format option seems to only work correctly for the root package. in my case the repo is shown there, but not in my workspaces
2. When i went through ALL workspaces & root and don't select any package the question "Run npm install to install new versions?" still is asked at the end

## Expected Behavior

For 1: The format option should work in every workspace project as well of course
For 2: If no version is selected at all the question should not appear.

Also: Is it possible to add a option to this library to skip the install question at the end and always install (if versions are selected)? I used [npm-check](https://github.com/dylang/npm-check) before and it installs right away and also installs only the changed packages by calculating a install command by the interactions of the user like for example `npm install package-a@1.0.1 package-b@2.9.3 ...`

**Comment by raineorshine** [comment]
Hi, thanks for reporting.

> 1. The format option seems to only work correctly for the root package. in my case the repo is shown there, but not in my workspaces

`--format repo` is working for me in workspaces:

```
ncu -ws --root -i --format repo                                                                              ✓
Upgrading /Users/raine/projects/ncu-issues/mono/package.json
[====================] 1/1 100%

✔ Choose which packages to update ›

No dependencies upgraded :)
Upgrading /Users/raine/projects/ncu-issues/mono/packages/some-local-app/package.json
[====================] 1/1 100%

? Choose which packages to update ›
  ↑/↓: Select a package
  Space: Toggle selection
  a: Toggle all
  Enter: Upgrade

❯ ◯ chalk  1.0.0  →  5.3.0    https://github.com/chalk/chalk
```

If you can provide a repo with steps to reproduce I am happy to investigate further.

Also note that not all repositories have a `repository` field specified in their package manifest, so those will appear blank.

> 2\. When i went through ALL workspaces & root and don't select any package the question "Run npm install to install new versions?" still is asked at the end

Good catch. Fixed and published in `v16.10.19`.

> Also: Is it possible to add a option to this library to skip the install question at the end and always install (if versions are selected)? I used [npm-check](https://github.com/dylang/npm-check) before and it installs right away and also installs only the changed packages by calculating a install command by the interactions of the user like for example `npm install package-a@1.0.1 package-b@2.9.3 ...`

Please open a separate issue for this one. I have some thoughts about it, but it would be better to discuss independently from the workspace issues.

**Comment by EinfachHans** [comment]
I finally find the time to do this, sorry for the delay...
Check the reproducable repo here: https://github.com/EinfachHans/ncu-format-bug

**Steps**:
1. Checkout
2. "npm i"
3. "npm run check:updates"

This will show the repo for the dependency in the root folder, after that it will not show it for the dependencies in the nested worskapce project

I also find the causing problem here, seems that the `install-strategy="shallow"` that i defined in the **.npmrc** is causing this. If you comment that line out, remove the **package-lock.json** and delete all node_modules and install again it works 🤔

**Comment by raineorshine** [comment]
Thanks so much, that makes it very clear.

The issue was that it was only looking in `process.cwd()` for node_modules (where it reads the repository field from the manifest). I added the workspace folder to its search path so it should work correctly now. It will fall back to the project root so it should work for either install strategy.

Published in `v16.12.3`. Let me know if it works!

**Comment by EinfachHans** [comment]
Works, thanks 😊
